eta/core/models.py

```diff
@@ -0,0 +1,909 @@
+    '''
+    model, models_dir, _ = _find_model(name)
+    if model.is_in_dir(models_dir):
+        _delete_model_from_dir(model, models_dir)
+
+
+def flush_old_models():
+    '''Deletes local copies of any old models, i.e. models for which the number
+    of versions stored on disk exceeds `eta.config.max_model_versions_to_keep`.
+
+    The models are not removed from their associated manifests and can be
+    downloaded again at any time.
+    '''
+    max_vers = eta.config.max_model_versions_to_keep
+    if max_vers < 0:
+        # No flushing required
+        return
+
+    # Get downloaded models
+    downloaded_models = _list_models(downloaded_only=True)[0]
+
+    # Group by base name
+    bmodels = defaultdict(list)
+    for model, mdir in itervalues(downloaded_models):
+        bmodels[model.base_name].append((model, mdir))
+
+    # Sort by version (newest first)
+    bmodels = {
+        k: sorted(v, reverse=True, key=lambda vi: vi[0].comp_version)
+        for k, v in iteritems(bmodels)
+    }
+
+    # Flush old models
+    for base_name, models_list in iteritems(bmodels):
+        num_to_flush = len(models_list) - max_vers
+        if num_to_flush > 0:
+            logger.info(
+                "*** Flushing %d old version(s) of model '%s'",
+                num_to_flush, base_name)
+            for model, models_dir in reversed(models_list[max_vers:]):
+                _delete_model_from_dir(model, models_dir)
+
+
+def flush_models_directory(models_dir):
+    '''Deletes the local copies of all models in the given models directory.
+
+    The models are not removed from their associated manifests and can be
+    downloaded again at any time.
+
+    Args:
+        models_dir: the models directory
+
+    Raises:
+        ModelError: if the directory contains no models manifest
+    '''
+    _warn_if_not_on_search_path(models_dir)
+    for model in ModelsManifest.from_dir(models_dir):
+        if model.is_in_dir(models_dir):
+            _delete_model_from_dir(model, models_dir)
+
+
+def flush_all_models():
+    '''Deletes all local copies of all models on the models search path.
+
+    The models are not removed from their associated manifests and can be
+    downloaded again at any time.
+    '''
+    for models_dir in etau.make_search_path(eta.config.models_dirs):
+        flush_models_directory(models_dir)
+
+
+def init_models_dir(new_models_dir):
+    '''Initializes the given directory as a models directory by creating an
+    empty models manifest file for it.
+
+    The directory is created if necessary.
+
+    Note that the directory is not automatically added to your models search
+    path, so models in this directory will not be findable until you update
+    your ETA config.
+
+    Args:
+        new_models_dir: the directory to initialize
+
+    Raises:
+        ModelError: if the models directory is already initialized
+    '''
+    if ModelsManifest.dir_has_manifest(new_models_dir):
```

Comment:
To not require a user to manually initialize the directory s/he has added to her/his config, we could add a bit of code to the eta initialization that call this for each models directory and just catches this Exception and moves on.